lib/WebService/Client.pm

```diff
@@ -103,6 +100,26 @@ sub log {
     $self->logger->$log_method($msg);
 }
 
+sub prepare_response {
+    my ($self, $res, $req, $num_attempts) = @_;
+    Moo::Role->apply_roles_to_object($res, 'HTTP::Response::Stringable');
+    return;
+}
```

Comment:
I think the primary use-case is adding your own roles to the HTTP::Response. E.g.:

```perl
if (not $res->is_success) {
    Moo::Role->apply_roles_to_object($res, 'My::Client::Error');
}
```